transcom/mymove
# B-17910 history audit log - services counselor creates or modifies a shipment

## [Agility B-17910](https://www13.v1host.com/USTRANSCOM38/story.mvc/Summary?oidToken=Story%3a867954)

## Summary

I have verified the items in the H&AL are correctly shown, and have added the downstream action of the closeout office added to the H&AL, as that is part of the flow for editing/creating a move for the SM.

## Verification Steps for the Author

These are to be checked by the author.


## Verification Steps for Reviewers
(Playwright test for the SM creating a move and the SC getting to the H&AL page not included)

These are to be checked by a reviewer.


### Setup to Run the Code

- [Instructions for starting the MilMove application](https://transcom.github.io/mymove-docs/docs/getting-started/application-setup/)
- [Instructions for running tests](https://transcom.github.io/mymove-docs/docs/getting-started/development/testing)

### How to test

1. Access the
2. Login as a service member
3. Create a ppm
4. Log in as SC
5. Navigate to the move that was newly created
6. Edit details of the move
7. Observe the changes populate properly in the audit history tab
8. Create a new PPM for the service member
9. Observe the changes populate in the audit history tab

To make it easier, you can run my automation steps for this:
https://gist.github.com/landan-parker-0/fb8d17ab88b79dabd56d83eb3591bec6
Just dump that into a file named scEditsServiceMemberPPM or something within the test folder: office, where the other playwright flows are.
It will show up as a test
You can run it with command line.
```yarn playwright test -g 'services counselor edits ppm'```


```ini
Status: SUBMITTED

Pro-gear: Yes

Pro-gear weight: 245 lbs

Spouse pro-gear weight: 357 lbs

Estimated weight: 602 lbs

Expected departure date: 17 Feb 2024

Sit expected: Yes

SIT estimated storage start: 17 Feb 2024

SIT estimated storage end: 24 Feb 2024

SIT estimated storage weight: 526 lbs

Pickup postal code: 63653

Destination postal code: 29101

SIT location: DESTINATION

SIT estimated cost: $104.09

PPM estimated incentive: $22,603.25
```

### Frontend


### Backend


### Database

The history fetcher sql file has changed
https://github.com/transcom/mymove/blob/2d9d1bbeb70cdb73a8283892c5265953b539e329/pkg/assets/sql_scripts/move_history_fetcher.sql
```javascript
move_logs AS (
  SELECT
    audit_history.*,
    jsonb_agg(jsonb_strip_nulls(
      jsonb_build_object(
        'closeout_office_name',
        (SELECT transportation_offices.name FROM transportation_offices WHERE transportation_offices.id = uuid(c.closeout_office_id))
      ))
    )::TEXT AS context,
    NULL AS context_id
  FROM
    audit_history
  JOIN move ON audit_history.object_id = move.id
  JOIN jsonb_to_record(audit_history.changed_data) as c(closeout_office_id TEXT) on TRUE
  WHERE audit_history.table_name = 'moves' group by audit_history.id
),

```


#### Any new migrations/schema changes:


## Screenshots
<img width="1057" alt="image" src="https://github.com/transcom/mymove/assets/148263228/a6333738-371b-43ef-92c3-ee8d202a7056">

<img width="640" alt="image" src="https://github.com/transcom/mymove/assets/148263228/9b4dbaa9-f6f5-4f74-a335-319cfaf34217">
![image](https://github.com/transcom/mymove/assets/148263228/c717cefe-41a7-40a5-9141-08c945dc4ed4)
